src/Orchard/FileSystems/AppData/AppDataFolder.cs

```diff
@@ -113,13 +113,35 @@ public class AppDataFolder : IAppDataFolder {
 
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
         public void CreateFile(string path, string content) {
-            using (var stream = CreateFile(path)) {
-                using (var tw = new StreamWriter(stream)) {
-                    tw.Write(content);
-                }
+            var filePath = CombineToPhysicalPath(path);
+            var folderPath = Path.GetDirectoryName(filePath);
+            if (!Directory.Exists(folderPath)) {
+                Directory.CreateDirectory(folderPath);
+            }
+
+            // for existing files, just overwrite
+            if (File.Exists(filePath)) {
+                File.WriteAllText(filePath, content);
+            } else {
+                WriteAllTextWithTempFileAndMove(filePath, content);
             }
         }
 
+        private static void WriteAllTextWithTempFileAndMove(string path, string content) {
+            // generate a temp filename
+            var tempPath = path + "." + Path.GetRandomFileName();
```

Comment:
This should be created in the environment temporary folder, just in case they are not deleted correctly and don't pollute the app_data.